src/rdpc/analysesProcessor.ts

```diff
@@ -69,6 +115,50 @@ export const analysisStream = async function* ({
   }
 };
 
+export const aggregateSpecimensByDonorId = (
+  analyses: AnalysisWithSpecimens[]
+): SpecimensByDonors => {
+  const result = analyses.reduce<SpecimensByDonors>((acc, analysis) => {
+    const donorWithSpecimens = analysis.donors.reduce<SpecimensByDonors>(
+      (_acc, donor) => {
+        _acc[donor.donorId] = donor.specimens;
+        return _acc;
+      },
+      {}
+    );
+
+    Object.entries(donorWithSpecimens).forEach(([donorId, specimens]) => {
```

Comment:
Why did we have to create donorWIthSpecimens? Could we simplify this by just having a single forEach loop on the `analysis.donors` array?

```
const result = analyses.reduce<SpecimensByDonors>((acc, analysis) => {
    analysis.donors.forEach(donor => {
        acc[donor.donorId] = acc[donor.donorId] ? donor.specimens : mergeSpecimens(donor.specimens, acc[donor.donorId]);
    });
    return acc;
}, {});
```

That's untested but the point is we should be able to do this in a single loop - not for efficiency (this will all be fast) but because split up like it is right now makes keeping track of what our data is quite tricky. Just loop through all donors, add their specimen to the accumulator, or merge with the existing list in the accumulator.

In either case, maybe some code comments could be helpful.